packages/daimo-api/src/api/getSwapRoute.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  BigIntStr,
+  EAccount,
+  getSupportedSendCoinByAddress,
+  now,
+  ProposedSwap,
+} from "@daimo/common";
+import { SwapRouter as SwapRouter02 } from "@uniswap/router-sdk";
+import { WETH9 } from "@uniswap/sdk-core";
+import { ADDRESS_ZERO } from "@uniswap/v3-sdk";
+import { Address, Hex, parseAbi } from "viem";
+
+import { UNISWAP_V3_02_ROUTER_ADDRESS } from "../network/uniswapClient";
+import { ViemClient } from "../network/viemClient";
+
+// Direct path length is: 0x (1) + Token (20) + Fee (3) + Token (20) = 44 * 2.
+const SINGLE_POOL_LENGTH_HEX = 88;
+
+// Given an exact amount input of tokenIn, retrieve the best-effort swap path
+// to get tokenOut.
+export async function getSwapQuote({
+  amountInStr,
+  tokenIn,
+  tokenOut,
+  fromAccount,
+  toAddr,
+  chainId,
+  vc,
+}: {
+  amountInStr: BigIntStr;
+  tokenIn: Address;
+  tokenOut: Address;
+  fromAccount: EAccount;
+  toAddr: Address;
+  vc: ViemClient;
+  chainId: number;
+}) {
+  const amountIn: bigint = BigInt(amountInStr);
+  const swapQuote = await vc.publicClient.readContract({
+    address: "0xD49Ad4D3945A655C508498571cFf092D261bf860", // TODO: config
+    abi: parseAbi([
+      "function quote(uint128 amountIn, address tokenIn, address tokenOut) public view returns (uint256 amountOut, bytes memory swapPath)",
```

Comment:
Should include contract Solidity and use wagmi ABI generation as we talked about